src/averageAges.js

```diff
@@ -36,8 +36,19 @@ function calculateMenAverageAge(people, century) {
  *
  * @return {number}
  */
+
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = people.filter((woman) => woman.sex === 'f'
+    && (withChildren === undefined
+    || people.find((motherName) => motherName.mother === woman.name)));
+
+  return getAverageAge(women);
+}
+
+function getAverageAge(genderList) {
+  return Math.round(genderList.reduce((sum, currentPerson) => sum
+    + currentPerson.died - currentPerson.born, 0)
+    / genderList.length * 100) / 100;
```

Comment:
Everything is written correctly now, but such a feature is difficult to read and maintain in the future.
Let's break it down into smaller parts.
```suggestion
function getAverageAge(people) {
  const totalAge = people.people((total, { died, born }) => total + died - born, 0);
  return +(totalAge / people.length).toFixed(2);
```